src/main/java/org/cirdles/topsoil/data/Dataset.java

```diff
@@ -33,5 +35,11 @@
     public List<Field<?>> getFields();
 
     public List<Entry> getEntries();
+    
+    public void removeEntry(Entry entry);
+    
+    public void addObserver(Chart Chart, VariableContext variableContext);
```

Comment:
`Dataset` shouldn't know about `Chart`. If you need a method like this, you need to introduce a new interface.